### PR TITLE
Reduce test warnings

### PR DESCRIPTION
**Pull request by samit4me**
PR for #909.

Reduce test warnings by passing required propTypes to the components.

To resolve the `<App />` test warning I added `babel-plugin-rewire` to mock the `Banner` so it is a string and not an object. This involved modifying the "internals" so this may need attention.

Resolving the `... ReactComponentTreeDevtool ...` warnings was a coincidence of extending the `` test coverage. This warning has been resolved in React `15.3.1`, see [React Issue 7240](https://github.com/facebook/react/issues/7240).

### CODE REVIEW

File: app/containers/App/tests/index.test.js [hunk 1]
```diff
@@ -1,11 +1,15 @@
...
+
 import expect from 'expect';
 import { shallow } from 'enzyme';
 import React from 'react';
 
-import App from '../index';
-import Footer from 'components/Footer';
-
 describe('<App />', () => {
+  /* eslint no-underscore-dangle: ["error", {"allow": ["__Rewire__", "__ResetDependency__"]}] */
```
Comment (mxstbr) [inline app/containers/App/tests/index.test.js]: Let's move this to the `package.json` and allow `__Rewire__` and `__ResetDependency__` globally – I have some other ideas where we could use rewire! (e.g. reducer initial state)

**Comment by mxstbr** [comment]
Thanks so much for taking the time to fix this!

I love that you included rewire, good idea 👌 Also a fan of the `renderComponent` pattern, makes the tests a lot cleaner!

One small nitpick then this is good to merge.

**Comment by samit4me** [comment]
Thank you for the positive feedback @mxstbr  😃 . 

Just re-read the [CONTRIBUTING](https://github.com/mxstbr/react-boilerplate/blob/master/.github/CONTRIBUTING.md) docs and noticed that PR's should be against the `dev` branch, would you like this PR changed? 

Unfortunately, this PR tests fail against the latest `dev` branch because **rewire** is not working. I'm not sure why but it's most likely got something to do with the dependency versions.

**Comment by gihrig** [comment]
@samit4me 
> Just re-read the CONTRIBUTING docs and noticed that PR's should be against the dev branch, would you like this PR changed?

Yes, this PR must be submitted against `dev`. Since it breaks `dev`, it makes merging dev into `master` a delayed debugging nightmare. Merging this PR to `master` breaks the whole idea of having a `dev` branch.

Sorry about that 😢

**Comment by samit4me** [comment]
No problems @gihrig , thank you for letting me know, what you explained makes allot of sense, thanks.

**Pull request closed without merging**